api/app/Http/Controllers/Api/ApiController.php

```diff
@@ -29,4 +31,19 @@ protected function errorResponse(string $message, int $status = JsonResponse::HT
             ]
         ], $status);
     }
+
+    public function createPaginatedResponse(
+        LengthAwarePaginator $paginator,
+        PresenterInterface $presenter
+    ): JsonResponse {
+        return new JsonResponse([
+            'data' => $presenter->presentCollection(collect($paginator->items())),
```

Comment:
Not all of the presenters have `presentCollection` method, so probably you need to add another interface extended from  `PresenterInterface` that will contain `presentCollection` method. Also, `collect()` should be moved to controller action, because in most cases the data already a collection.

Reply:
It's so, but `$paginator->items()`, returns array, and for transforming array to Collection, i used it.